miio/gateway.py

```diff
@@ -252,9 +265,18 @@ def discover_devices(self):
             DeviceType.D1WallSwitchTripleNN: D1WallSwitchTripleNN,
             DeviceType.ThermostatS2: ThermostatS2,
         }
-        devices_raw = self.get_prop("device_list")
         self._devices = {}
 
+        # Skip the models which do not support getting the device list
+        if self.model == GATEWAY_MODEL_EU:
+            _LOGGER.info(
```

Comment:
Use `warning` instead of `info` for these warnings, this is otherwise good to go.

Reply:
The internall method will not help much becuase you would still need to check for the return value of that method and then have some specific error message to specify which function is not supported.
So do not think adding such method will actually reduce the amount of code...

Reply:
Fair enough, let's merge this then!